background_scripts/idem_bg.js

```diff
@@ -1,266 +1,353 @@
+  return fetch(`${TOC_URL}/api${path}`, {
+    method,
+    headers: { 'Content-Type': 'application/json' },
+    body: !!payload ? JSON.stringify(payload) : undefined
+  }).then(response => response.json())
+}
 
 const actions = {
+  log_activate: function () {
+    _log = true
+    if (!_tabId) return
+    browser.tabs.sendMessage(_tabId, {
+      command: 'log_activate'
+    })
+  },
 
-	log_activate: function() {
-		_log = true
-		if (!_tabId) return
-		browser.tabs.sendMessage(_tabId, {
-			command: "log_activate",
-		})
-	},
+  log_deactivate: function () {
+    _log = false
+    if (!_tabId) return
+    browser.tabs.sendMessage(_tabId, {
+      command: 'log_deactivate'
+    })
+  },
 
-	log_deactivate: function() {
-		_log = false
-		if (!_tabId) return
-		browser.tabs.sendMessage(_tabId, {
-			command: "log_deactivate",
-		})
-	},
+  log_get_status: function () {
+    return _log
+  },
 
-	log_get_status: function() {
-		return _log
-	},
+  tab_get_id: function () {
+    return _tabId
+  },
 
-	tab_get_id: function() {
-		return _tabId
-	},
+  tab_initialize: function () {
+    // init for this tab
+    browser.tabs.query({ currentWindow: true, active: true }).then(tabs => {
+      _tabId = tabs[0].id
+      _ide = SKIP_IDE_DETECTION || tabs[0].url.includes('codingame.com/ide/')
+      set_idem_logo({ tabId: _tabId })
+    });
+    // get and save CG id
+    ensureScriptLoaded().then(() => {
+      browser.tabs.sendMessage(
+        _tabId,
+        {
+          command: 'tab_find_id'
+        },
+        function (r) {
+          if (r.res === 'obj') _cg_id = r.obj
+          else _cg_id = undefined
+          // notify popup. This handle asynchronus problem of Promise
+          browser.runtime.sendMessage({
+            command: 'cg_id_change',
+            obj: _cg_id
+          })
+        }
+      )
+    })
+  },
 
-	tab_initialize: function() {
-		browser.tabs.query({currentWindow: true, active: true})
-			.then(tabs => {
-				_tabId = tabs[0].id
-				_ide = tabs[0].url.includes('codingame.com/ide/')
-				set_idem_logo({tabId: _tabId})
-			})
-	},
+  tab_goto: function () {
+    if (_tabId) {
+      // highlight to keep current tab visible ?
+      browser.tabs.update(_tabId, { highlighted: true })
+    }
+  },
 
-	tab_goto: function() {
-		if (_tabId) {
-			// highlight to keep current tab visible ?
-			browser.tabs.update(_tabId, {highlighted: true})
-		}
-	},
+  get_cg_id: function () {
+    return _cg_id
+  },
 
-	action_get_buffer: function() {
-		return _buffer
-	},
+  action_get_buffer: function () {
+    return _buffer
+  },
 
-	twitch_connect: function(channel) {
-		const status = _twitch?.status
-		if (status == 'CONNECTING' || status === 'OPEN') {
-			return `Already connected to channel [${_twitch.channel}]`
-		} else {
-			if (!channel) return 'Empty channel'
-			_twitch = new TwitchWS(channel, actions)
-			update_idem_logo()
-		}
-	},
+  search_tournament: function () {
+    console.log('Searching tournaments...')
```

Comment:
Rajouter la garde du logger. Je copie la technique de Julien avec l'emoji 👋 pour les autres
```suggestion
    if (_log) console.log('Searching tournaments...')
```